### PR TITLE
Meta Quest 2 / 3 support

### PR DESCRIPTION
**Pull request by RevoluPowered**
Replaces #193 

### What does this do:
- Major engine upgrade to a much newer version of Godot Engine, with the fixes we helped with on the meta quest 3. (https://github.com/the-mirror-gdp/godot/tree/gordon/rebase-rc-2 to be merged) 
- Adds performance advantages for quest2/quest3 and low end hardware.
- In theory fixes the intel crashing issues thanks to us removing our blur shader for the UI. (it was a huge performance hit)
- Allows the Game UI to be used from within VR, renders all existing UI. (Doesn't allow the build menu but this was out of scope for this work)
- Ensures the player is not visible for themselves on the local client, this will ensure no weird clipping happens.
- The mirror can run in standalone mode on the quest 2 and quest 3.
- Android doesn't flicker anymore. (Thanks to the godot team for their amazing help ❤️)  https://github.com/godotengine/godot/pull/94289
- GameUI had to be moved to GameUI.instance as GameUI.instance is set based on if you are using VR or if you are using the Desktop UI, they are very different. GameUI.instance is useful because it ensures that you point at the correct UI and that you don't need to delete/readd the game UI, this was a major refactor and hugely painful to get working, there may be issues we do not know about without wider testing.
- Moves all settings across to our user settings, things that don't have a setting are currently disabled (this is a bug and should be fixed before merge)
- Forces glow and various graphical settings off when using VR, this is because it must be disabled officially any post processing except MSAA should be disabled for VR, baring foveation and some settings under OpenXR.
- Removes the GameUI autoload and replaces it with the GameUI singleton using GameUI.instance.

### Bugs noticed in testing the builds in the PR:
- [ ] Project Quality settings need to be fixed the quality settings are forcing low quality details.
- [ ] Ensure I do not store the empty test account details in the UI elements (purposefully used an empty account so it could be deleted later

### CODE REVIEW

File: mirror-godot-app/art/environments/physical_clouds_sky.tres
Comment (RevoluPowered) [inline mirror-godot-app/art/environments/physical_clouds_sky.tres]: These need to be contextual for VR vs Desktop

File: mirror-godot-app/art/environments/physical_sky.tres
Comment (RevoluPowered) [inline mirror-godot-app/art/environments/physical_sky.tres]: These need to be contextual for VR vs Desktop

File: mirror-godot-app/creator/selection/inspector/categories/inspector_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/creator/selection/inspector/categories/inspector_environment.gd]: This needs to be re-enabled for desktop.

File: mirror-godot-app/player/cameras/camera_manager.gd
Comment (RevoluPowered) [inline mirror-godot-app/player/cameras/camera_manager.gd]: These should be contextual and not used when VR is enabled.

File: mirror-godot-app/player/cameras/camera_manager.gd
Comment (RevoluPowered) [inline mirror-godot-app/player/cameras/camera_manager.gd]: This should be fixed as it was removed for a test

File: mirror-godot-app/player/vr/vr_controller.gd
Comment (RevoluPowered) [inline mirror-godot-app/player/vr/vr_controller.gd]: remove print

File: mirror-godot-app/player/vr/vr_controller.gd
Comment (RevoluPowered) [inline mirror-godot-app/player/vr/vr_controller.gd]: remove print

File: mirror-godot-app/player/vr/vr_controller.tscn
Comment (RevoluPowered) [inline mirror-godot-app/player/vr/vr_controller.tscn:17]: TODO check emoji menu works

File: mirror-godot-app/prefabs/space/templates/space_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_environment.gd]: Re-enable

File: mirror-godot-app/prefabs/space/templates/space_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_environment.gd:97]: Ensure the default GLOW is never enabled nor any of the other features in the new world environment, they should be set AFTER construction not before in the defaults.

File: mirror-godot-app/prefabs/space/templates/space_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_environment.gd]: Enable me

File: mirror-godot-app/prefabs/space/templates/space_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_environment.gd]: Enable me

File: mirror-godot-app/prefabs/space/templates/space_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_environment.gd]: Enable me

File: mirror-godot-app/prefabs/space/templates/space_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_environment.gd]: fix me

File: mirror-godot-app/prefabs/space/templates/space_environment.gd
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_environment.gd]: Enable me

File: mirror-godot-app/prefabs/space/templates/space_template.tscn
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_template.tscn]: This should later be configured via the space template gd NOT by the defaults, in VR this will nuke performance

File: mirror-godot-app/prefabs/space/templates/space_template.tscn
Comment (RevoluPowered) [inline mirror-godot-app/prefabs/space/templates/space_template.tscn]: This should be the new default but shadows should be enabled by code based on a setting.

File: mirror-godot-app/project.godot
Comment (RevoluPowered) [inline mirror-godot-app/project.godot:571]: These settings MUST be moved to our game settings handler. there is no way we can leave them part of the defaults.

File: mirror-godot-app/project.godot
Comment (RevoluPowered) [inline mirror-godot-app/project.godot:631]: These should be enabled for all VR headsets, we should ensure we disable OpenXR entirely on Mac for the future until they can provide a non crashing desktop experience with OpenXR enabled.

File: mirror-godot-app/scripts/autoload/bootup.gd
Comment (RevoluPowered) [inline mirror-godot-app/scripts/autoload/bootup.gd:107]: should be setter

File: mirror-godot-app/scripts/autoload/util_funcs.gd
Comment (RevoluPowered) [inline mirror-godot-app/scripts/autoload/util_funcs.gd]: This is a problematic function. It actually makes more sense to remove this and make a new behaviour. i.e. IF build is standlone statically enable this, if build is for windows/linux enable this, if macos, forcefully disable via the .cfg so this setting is literally something that should be part of our .cfg and read from there, nowhere else.

File: mirror-godot-app/ui/common/shaders/dark_blue_blur.mat.tres
Comment (RevoluPowered) [inline mirror-godot-app/ui/common/shaders/dark_blue_blur.mat.tres:7]: We need to completely depreciate use of the blur shader, it causes problems on intel and on the quest 2/3

File: mirror-godot-app/ui/common/shaders/ui_blurr.gdshader
Comment (RevoluPowered) [inline mirror-godot-app/ui/common/shaders/ui_blurr.gdshader:5]: clarify we are removing this and delete this

File: mirror-godot-app/ui/game/GameUIStatic.gd
Comment (RevoluPowered) [inline mirror-godot-app/ui/game/GameUIStatic.gd]: dead code

File: mirror-godot-app/ui/main_menu/common/space_panel.gd
Comment (RevoluPowered) [inline mirror-godot-app/ui/main_menu/common/space_panel.gd:10]: ask for liza to test the panels and check they are correct and working

AFAIK these values always error when used even when uncommented and this doesn't work but we might want to remove this change.

File: mirror-godot-app/ui/notifications/notify.gd
Comment (RevoluPowered) [inline mirror-godot-app/ui/notifications/notify.gd:10]: This code should really check if notifications should be enabled, since we essentially removed what the original intent was doing.

File: scripts/compile.sh
Comment (RevoluPowered) [inline scripts/compile.sh]: rename to compile-android but update the details and comments

**Comment by RevoluPowered** [comment]
- MacOS build cannot be opened....


Fixed by renaming the executable to The Mirror Dev

**Comment by RevoluPowered** [comment]
We need to remove the debug shapes I added to the world for renderer tests.

File: mirror-godot-app/scripts/autoload/util_funcs.gd
Comment (RevoluPowered) [reply mirror-godot-app/scripts/autoload/util_funcs.gd]: Leaving comment open so we can sort this after merge.

**Review by Kluskey — APPROVED** [review]
Screenshot-2024-07-25-000727@2x
Approving assuming your comments are resolved. Thank you for your hard work on this!!

**Comment by RevoluPowered** [comment]
> Screenshot-2024-07-25-000727@2x Approving assuming your comments are resolved. Thank you for your hard work on this!!

Yeah the issue with the Mac build was just the naming on my mac, if you rename The Mirror Dev to The Mirror Dev 3 it wont launch

**Pull request merged**
